### ISSUE TITLE
Can't connect to replica set [MongoDB 3.0.8 / Robomongo 1.0.0]

### ISSUE THREAD

**Issue by pascal2944** [Studio3T/robomongo#1345]
Hello,

I have installed Robomongo 1.0.0 and try to connect to a replica set but it doesn't work.

I create separated Connection with the 3 members (Direct Connection without SSH or SSL) and I can connect to each members with no troubles. 

But when I create the Connection for the replica set including the same 3 members, I always have the error message:

"Cannot connect to replica set...
Set's primary is unreachable.
Reason:
No member of the set is reachable."

In the Logs windows I can read this:
Error: Establish connection failed. No member of the set is reachable.. Connection: Replicaset vpremgo

Thanks for your help.

**Comment by simsekgokhan** [comment]
Hi @pascal2944 , thanks for reporting the problem.

Can you share more info about your replica set is it local or cloud service?
Can you also share your connection settings details for replica set connection? Are you using SSH or SSL etc.. Please also try to connect one more time with replica set name entered on UI.

**Comment by pascal2944** [comment]
Hello,
No, it's not in local, the database is on 3 different servers (linux). I don't use SSH or SSL

Here is my connection setting for the replica set:
  {
   "connectionName" : "Replicaset vpremgo",
   "credentials" : [
    {
     "databaseName" : "admin",
     "enabled" : true,
     "mechanism" : "SCRAM-SHA-1",
     "userName" : "my_admin_user",
     "userPassword" : "my_admin_user_pwd"
    }
   ],
   "defaultDatabase" : "",
   "isReplicaSet" : true,
   "replicaSet" : {
    "0" : "vpremgo03:27017",
    "1" : "vpremgo04:27017",
    "2" : "vpremgo05:27017",
    "cachedSetName" : "",
    "readPreference" : 0,
    "setNameUserEntered" : "rs_my_replicaset"
   },
   "serverHost" : "vpremgo03",
   "serverPort" : 27017,
   "ssh" : {
    "askPassword" : false,
    "enabled" : false,
    "host" : "",
    "method" : "publickey",
    "passphrase" : "",
    "port" : 22,
    "privateKeyFile" : "",
    "publicKeyFile" : "",
    "userName" : "",
    "userPassword" : ""
   },
   "ssl" : {
    "allowInvalidCertificates" : false,
    "allowInvalidHostnames" : false,
    "askPassphrase" : false,
    "caFile" : "",
    "crlFile" : "",
    "pemKeyFile" : "",
    "pemPassPhrase" : "",
    "sslEnabled" : false,
    "useAdvancedOptions" : false,
    "usePemFile" : false
   },
   "uuid" : "{c959c22d-2b0f-401e-8418-ad245495770a}"
  },

I have also try the ip address instead of the hostname, I have set the name of the replica on UI.

Here is the connection setting of the first member I can connect in "Direct Connection":

  {
   "connectionName" : "vpremgo03",
   "credentials" : [
    {
     "databaseName" : "admin",
     "enabled" : true,
     "mechanism" : "SCRAM-SHA-1",
     "userName" : "my_admin_user",
     "userPassword" : "my_admin_user_pwd"
    }
   ],
   "defaultDatabase" : "",
   "isReplicaSet" : false,
   "serverHost" : "vpremgo03",
   "serverPort" : 27017,
   "ssh" : {
    "askPassword" : false,
    "enabled" : false,
    "host" : "",
    "method" : "publickey",
    "passphrase" : "",
    "port" : 22,
    "privateKeyFile" : "",
    "publicKeyFile" : "",
    "userName" : "",
    "userPassword" : ""
   },
   "ssl" : {
    "allowInvalidCertificates" : false,
    "allowInvalidHostnames" : false,
    "askPassphrase" : false,
    "caFile" : "",
    "crlFile" : "",
    "pemKeyFile" : "",
    "pemPassPhrase" : "",
    "sslEnabled" : false,
    "useAdvancedOptions" : false,
    "usePemFile" : false
   },
   "uuid" : "{86820f6d-dd8e-4a04-8ee0-f44ce5a83f14}"
  },

Thanks.

**Comment by simsekgokhan** [comment]
Thanks a lot for details. I did not see any problem at first glance.

Can you also connect to a set member, run `rs.status()` and double check the set name and the all member names match with Robomongo connection settings?

![screenhunter_110 may 12 11 58](https://cloud.githubusercontent.com/assets/19390749/25991001/74c89dc8-370a-11e7-9bb0-e277547156f8.jpg)

**Comment by pascal2944** [comment]
I have run rs.status() from vpremgo03 (Direct Connection) and for the key "set" I can read "rs_my_replicaset" but for the key "name" it's the ip address corresponding to eth2 of the server while I can only connect to this server thru eth1.
May be it's more a trouble with the way robomongo use the network (vlan, firewall...) to connect to the replica ?

Thanks.

**Comment by simsekgokhan** [comment]
Yes, that must be the problem. The member names in rs.status(), must be used without change in Robomongo connections settings for replica sets. The workaround can be using port forwarding on terminal but you will have to do forwarding for each member. We can help with that if you decide that option. We have how-to steps for windows, linux and mac port forwarding.

Another workaround is using SSH & Replica Sets, but this combination is currently not working properly, it will/must be re-implemented.

**Comment by simsekgokhan** [comment]
Another option, you can try to connect to Mongo Atlas 3.4 free cluster with the latest beta: [Robomongo 1.1 - Beta version with MongoDB 3.4 Support](https://github.com/Studio3T/robomongo/issues/1250#issuecomment-301036770)

**Comment by pascal2944** [comment]
Thanks, I have try SSH & Replica Sets but it's the same trouble:

Info: Creating SSH tunnel to x.x.x.x:22...                        => thru eth1
Info: SSH tunnel created successfully
Info: Connecting to Replicaset vpremgo [Replica Set]x.x.x.x:27017...                        => thru eth1
Info: Replica set's cached set name cleared. Using user entered set name.
Error: Establish connection failed. No member of the set is reachable.. Connection: Replicaset vpremgo

I will see what is Mongo Atlas 3.4 free cluster.

**Comment by simsekgokhan** [comment]
Sure, Mongo Atlas is the official cloud database service from MongoDB. Basically, you can create and use 3.4 free cluster (replica set) with a couple of clicks after registering. Please see: https://www.mongodb.com/cloud/atlas

**Comment by pascal2944** [comment]
Thank you but we can't use your service Mongo Atlas for our databases.

**Comment by simsekgokhan** [comment]
You cannot use Mongo Atlas due to a connection problem (we can help) or other reasons?

**Comment by kpuscas** [comment]
I'm having a similar problem with 1.1.1 connecting to a ObjectRocket 3 node replica set. I get the error message "Set's primary is unreachable." I've also tried connecting to the primary via DirectConnect but that results in an auth error. I have connected to the replica set via python and java drivers and I have connected to the primary with Mongodb Compass.

**Comment by jupe** [comment]
I'm having this as well with 1.1.1 :/ . Anybody investigating or fixing this ?

**Comment by EvAlex22** [comment]
If user is defined with low permission  role like read or readWrite and tries to connect to RS the **replSetGetStatus** are issued by RoboMongo and fails.

**Comment by brzozowski** [comment]
I have managed to resolve the problem by adding my **current IP address** to the **IP Whitelist**.

![pobrane](https://user-images.githubusercontent.com/33604424/58291346-60edd100-7dbd-11e9-9647-6b2eb4b0c744.png)

**Comment by javaHelper** [comment]
Can someone please guide here: https://stackoverflow.com/questions/52945978/unable-to-connect-to-mongodb-replica-set-from-other-server-using-robo3t-and-in-c ?
